packages/protoplugin/src/ecmascript/gencommon.ts

```diff
@@ -59,7 +59,12 @@ export function makeFilePreamble(
   writeLeadingComments(file.getSyntaxComments());
   builder.push(`// @generated by ${pluginName} ${pluginVersion}`);
   if (parameter !== undefined) {
-    builder.push(` with parameter "${parameter}"`);
+    // rewrite_imports can be noisy and is more of an implementation detail
+    // so we strip it out of the preamble
+    const stripped = stripParameter(parameter, "rewrite_imports");
```

Comment:
It doesn't seem quite right to parse the "parameter" here. It is just a convention that options are delimited with a comma, and I think it's better to have just one piece of code dealing with the ambiguitiy.

We already parse the parameter in `createEcmaScriptPlugin()`. Let's just return the parameter we want to render there, pass it as a new argument to `createSchema()` (I'd suggest the name `pluginParameter`), and then to `createGeneratedFile()`.

This should be fairly clean, and we won't stumble over a user-defined plugin option that happens to start with "rewrite_imports".

Reply:
The parameter is actually already passed to `createSchema` along with other parameters.  Do you mean that this new `pluginParameter` argument represents a string of the plugin parameters we want to render in the preamble?

Reply:
Yes.